net/WebSocketHandler.hpp

```diff
@@ -83,7 +83,7 @@ class WebSocketHandler : public ProtocolHandlerInterface
 #endif
         _shuttingDown(false)
         , _isClient(isClient)
-#ifdef ENABLE_DEBUG
+#if !MOBILEAPP
         , _unit(UnitBase::get())
```

Comment:
This boggles the mind - why do we have a member in each instance that stores the result of a static function - that seems unbelievable =)

Ash - what is going on here:

commit a89ed3b5350a55d265d5ea0131f3a342e28384ff
Author: Ashod Nakashian <ashod.nakashian@collabora.co.uk>

And why can't we use the global variable; that's really under-explained; also see Patrick's point on the #ifdef FOO.

Ideally we would have a git hook to check for #ifdefs and stop them creeping in in error =)

Reply:
Good question! Not obvious, but we now support running multiple tests in the same process. To do that, we have to create a new instance of the UnitX. The problem is that we have all sorts of threads and async logic and re-using the static version sends events/data to the wrong test (when we "switch" from one test to the next).

Multi-test support makes it far easier to add new variations on a given test-case (in the same unit-test files, no Makefile changes), faster build (no new translation units), and faster runtime (same process, same Kit fork). It's wins all around!

(The fact that `ENABLE_DEBUG` was always true is neither here nor there; tests are _explicitly_ disabled outside of debug builds.)